app/src/main/java/com/example/firstapplication/MainActivity.java

```diff
@@ -35,6 +35,8 @@ public void onClick(View v) {
                 viewPassword.setText( editPassword.getText().toString());
 
                 Intent intent = new Intent(MainActivity.this, HomePageActivity.class);
+                intent.putExtra("username",editUsername.getText().toString());
+                intent.putExtra("email", editEmail.getText().toString());
```

Comment:
Create a class called ConstantUtils and put in static variables for "email" and "username" so you can reference them anywhere you need them instead of typing them out. This will help you avoid possible bugs where the other end is not able to get the data sent from here due to a spelling mistake.

Reply:
okay I will do that